.github/workflows/terraform.yaml

```diff
@@ -0,0 +1,156 @@
+  AWS_SECRET_ACCESS_KEY: "${{ secrets.AWS_SECRET_ACCESS_KEY }}"
+  VASTCLOUD_NOTTY: 1
+
+jobs:
+  vast_on_aws:
+    name: "VAST ON AWS"
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        working-directory: ./cloud/aws
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Terraform Format
+        run: terraform fmt -check -recursive -diff
+
+      - name: "Create config file"
+        if: ${{ !env.ACT }}
+        run: |
+          echo "vast_peered_vpc_id = $VAST_PEERED_VPC_ID" >> .env
+          echo "vast_cidr = $VAST_CIDR" >> .env
+          echo "vast_aws_region = $VAST_AWS_REGION" >> .env
+
+      - name: "Substitute backend with Terraform Cloud"
+        run: |
+          cat<<EOF > step-1/backend.tf
+          terraform {
+            cloud {
+              organization = "$TF_ORGANIZATION"
+              token        = "$TF_API_TOKEN"
+              workspaces {
+                name = "$TF_WORKSPACE_PREFIX-step-1"
+              }
+            }
+          }
+          EOF
+          cat<<EOF > step-2/backend.tf
+          terraform {
+            cloud {
+              organization = "$TF_ORGANIZATION"
+              token        = "$TF_API_TOKEN"
+              workspaces {
+                name = "$TF_WORKSPACE_PREFIX-step-2"
+              }
+            }
+          }
+          EOF
+
+      - name: Deploy
+        id: deploy
+        continue-on-error: true
+        run: ./vast-cloud deploy --auto-approve
+
+      - name: Retry deploy
+        if: steps.deploy.outcome=='failure'
+        run: |
+          echo "Deploy sometimes fails for an unexplained reason."
+          echo "Retrying after a few minutes usually works."
+          sleep 200
+          ./vast-cloud deploy --auto-approve
+
+      - name: Start and restart VAST server
+        run: |
+          echo "Run start-vast-server"
+          ./vast-cloud start-vast-server
+
+          echo "Run start-vast-server again"
+          ./vast-cloud start-vast-server 2> /dev/null \
+            && { echo "Starting server again should fail"; false; } \
+            || true
+          [[ $(./vast-cloud get-vast-server | wc -w) = "1" ]] \
+            || { echo "Only one task should be started"; false; }
+
+          echo "Run restart-vast-server"
+          ./vast-cloud restart-vast-server
+          [[ $(./vast-cloud get-vast-server | wc -w) = "1" ]] \
+            || { echo "Only one task should be started"; false; }
+
+          echo "The task needs a bit of time to boot, sleeping for a while..."
+          sleep 100
+
+      - name: Test db empty from Lambda
+        run: |
+          result=$(./vast-cloud run-lambda -c "vast count")
+          echo "Expected vast count 0, got $result"
+          [[ $result = "0" ]]
+
+      - name: Import data
+        run: |
+          DATA_URL=https://raw.githubusercontent.com/tenzir/vast/master/vast/integration/data/suricata/eve.json
```

Comment:
This URL always points to master. Can you make it point to the current commit hash instead? If you're running in a `workflow_dispatch` context, this should work:

```
DATA_URL=https://raw.githubusercontent.com/tenzir/vast/blob/${{ github.sha }}/vast/integration/data/suricata/eve.json
```

Reply:
I think that `https://raw.githubusercontent.com/tenzir/vast/blob/${{ github.sha }}` will have the drawback of not existing in a fork. If we go this way, I propose `https://raw.githubusercontent.com/${{ github.repository }}/blob/${{ github.sha }}`.

Will this not work if we are not in `workflow_dispatch` ? Isn't the commit always published when the workflow is executed, independently from the type of trigger?

Reply:
Some triggers like `issue_comment` don't have a SHA attached.

+1 on embedding the repo as well.

Reply:
Actually, could we rather set this as an immutable external resource, for instance on S3 or a pinned commit? Depending on the Github environment complicates things if we want the tests to run from other places (like locally...)

Reply:
(also, the right URL seems to be `https://raw.githubusercontent.com/tenzir/vast/${{ github.sha }}` and not `https://raw.githubusercontent.com/tenzir/vast/blob/${{ github.sha }}` )

Reply:
> Actually, could we rather set this as a hardcoded immutable external resource, for instance on S3 or a pinned commit? Depending on the Github environment complicates things if we want the tests to run from other places (like locally...)

Personally I'd prefer to keep all things related to the CI in the CI. You could also in-line the file contents in the workflow, given that it's only 7 lines.

Reply:
The issue here is that the data needs to be accessed from the cloud. Here we are asserting both that:
- we can access the public internet
- we can load then read the 7 lines of data

Inlining the content in the workflow does not solve that. This is why we were using `raw.githubusercontent.com` in the first place, because it is publicly accessible.
- The most simple solution I see for now is to pin to a specific version (say v1.0.0) that is immutable and will be kept around "indefinitely".
- We could create a cloud resource within the stack and upload the data for the duration of the script. But this would require quite some plumbing to set the access rights correctly. I think this should be considered if at some point we integrate such a resource to the stack for some other purpose.
- We could create a cloud resource independent from the stack that hosts this data, but that will not bring much value over using a pined version on `raw.githubusercontent.com`